Contents/Code/siteToughLoveX.py

```diff
@@ -0,0 +1,116 @@
+import PAsearchSites
+import PAactors
+
+def search(results,encodedTitle,title,searchTitle,siteNum,lang,searchByDateActor,searchDate,searchSiteID):
+    if searchSiteID != 9999:
+        siteNum = searchSiteID
+    searchString = searchTitle.replace(" ","-").replace(",","").replace("'","").replace("?","").lower().strip()
+    Log("searchString " + searchString)
+    modelsPageSortByLetter = PAsearchSites.getSearchSearchURL(siteNum) + searchTitle[:1] #here we get first letter of searchTitle
+    actorResults = HTML.ElementFromURL(modelsPageSortByLetter)
+    actorPage = actorResults.xpath('//a[contains(@href,"' + searchString + '")]')[0].get('href') # looking for our model
+    searchResults = HTML.ElementFromURL(actorPage) # scene search is carried out through the model page
+
+    for searchResult in searchResults.xpath('//div[@class="content-box"]'):
+        titleNoFormatting = searchResult.xpath('.//h2//span')[0].text_content().strip()
+        sceneUrl = searchResult.xpath('.//a')[0].get('href')
+        curID = sceneUrl.replace('/','_').replace('?','!')
+        scenePage = HTML.ElementFromURL(sceneUrl) # geting releaseDate from scene page
+        releaseDate = parse(scenePage.xpath('//span[@class="date"]')[0].text_content().strip()).strftime('%Y-%m-%d')
+        score = 100 - Util.LevenshteinDistance(searchDate, releaseDate)
+        results.Append(MetadataSearchResult(id = curID + "|" + str(siteNum), name = titleNoFormatting + " [" + PAsearchSites.getSearchSiteName(siteNum) + "] " + releaseDate, score = score, lang = lang))
+    return results
+
+def update(metadata,siteID,movieGenres,movieActors):
+    pageURL = str(metadata.id).split("|")[0].replace('_', '/').replace('!','?')
+    Log('scene url: ' + pageURL)
+    detailsPageElements = HTML.ElementFromURL(pageURL)
+    art = []
+    # Summary
+    siteName = PAsearchSites.getSearchSiteName(siteID)
+    metadata.studio = siteName
+    try:
+        metadata.summary = detailsPageElements.xpath('//div[@class="description"]//p')[0].text_content().strip()
+        metadata.title = detailsPageElements.xpath('//h1')[0].text_content().strip()
+    except:
+        pass
+
+    # Director
+    metadata.directors.clear()
+    director = metadata.directors.new()
+    director.name = "Charles Dera"
+
+    # Collections / Tagline
+    metadata.collections.clear()
+    metadata.tagline = siteName
+    metadata.collections.add(siteName)
+
+    # Genres
+    try:
+        genres = detailsPageElements.xpath('//dd[2]')
```

Comment:
Maybe use this as a collection instead, and pull genres from //meta@name="keywords"? Stripping out stuff like "tough love", "karl toughlove", "HD videos", etc.

Reply:
here I would prefer not to add genres at all, because genres contain a lot of garbage  (_**tough love,karl toughlove,slut training,charles dera,top pornstars**_,father karl,_**best blowjobs,porn videos,HD videos**_,throat fucking)
how your opinion?

Reply:
Makes sense! Or perhaps just add something like:
`movieGenres.addGenre(Hardcore)`